R/stat-contour.r

```diff
@@ -31,26 +43,23 @@ stat_contour <- function(mapping = NULL, data = NULL,
 #' @usage NULL
 #' @export
 StatContour <- ggproto("StatContour", Stat,
-  required_aes = c("x", "y", "z"),
-  default_aes = aes(order = ..level..),
+                       required_aes = c("x", "y", "z"),
+                       default_aes = aes(order = ..level..),
 
-  compute_group = function(data, scales, bins = NULL, binwidth = NULL,
-                           breaks = NULL, complete = FALSE, na.rm = FALSE) {
-    # If no parameters set, use pretty bins
-    if (is.null(bins) && is.null(binwidth) && is.null(breaks)) {
-      breaks <- pretty(range(data$z), 10)
-    }
-    # If provided, use bins to calculate binwidth
-    if (!is.null(bins)) {
-      binwidth <- diff(range(data$z)) / bins
-    }
-    # If necessary, compute breaks from binwidth
-    if (is.null(breaks)) {
-      breaks <- fullseq(range(data$z), binwidth)
-    }
+                       compute_group = function(data, scales, bins = NULL, binwidth = NULL,
+                                                breaks = breaks_default,
+                                                complete = FALSE, na.rm = FALSE) {
+                         # Check is.null(breaks) for backwards compatibility
```

Comment:
I think we've gone off the rails a bit here because I wasn't clear exactly what I was thinking. I'd prefer not to make such a big change and instead of `breaks` is a function, call it like `fullseq` with the range of the data and the binwidth.